### PR TITLE
cli: add completion support for fish shell

### LINKED ISSUE #11283: Fish completion support
## Proposal / RFE

**Is your feature request related to a problem?**
Nope

**Describe the solution you'd like**
It would be nice to have CLI completion support for fish shell

### PR DESCRIPTION
**Pull request by sayboras**
Fixes: https://github.com/cilium/cilium/issues/11283

image

```release-note
Add completion support for fish shell
```

### CODE REVIEW

**Comment by qmonnet** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

File: cilium/cmd/root.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ func init() {
 	flags.StringP("host", "H", "", "URI to server-side API")
 	viper.BindPFlags(flags)
 	rootCmd.AddCommand(newCmdCompletion(os.Stdout))
-	rootCmd.SetOut(os.Stderr)
+	rootCmd.SetOut(os.Stdout)
```
Comment (qmonnet) [inline cilium/cmd/root.go:65]: Is this change intentional? I wonder if it might be the cause for the CI failure?

Comment (sayboras) [reply cilium/cmd/root.go:65]: This change is required as internally _cilium __complete_ command will be called, and it's using the stdout for showing completion options. 

I just checked the failure, seems like we are expecting help output in stderr. Let me see if there is easy way to fix this.

**Comment by sayboras** [comment]
test-me-please

**Comment by qmonnet** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]

**Pull request merged**